packages/web-components/src/checkbox/checkbox.template.ts

```diff
@@ -27,24 +28,13 @@ const indeterminateIndicator = html.partial(`
 export function checkboxTemplate<T extends Checkbox>(options: CheckboxOptions = {}): ElementViewTemplate<T> {
   return html<T>`
     <template
-      role="checkbox"
-      aria-checked="${x => (x.indeterminate ? 'mixed' : x.checked)}"
-      aria-required="${x => x.required}"
-      aria-disabled="${x => x.disabled}"
-      tabindex="${x => (x.disabled ? null : 0)}"
-      @keypress="${(x, c) => x.keypressHandler(c.event as KeyboardEvent)}"
+      tabindex="${x => (!x.disabled ? 0 : void 0)}"
       @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
+      @input="${(x, c) => x.inputHandler(c.event as Event)}"
+      @keypress="${(x, c) => x.keydownHandler(c.event as KeyboardEvent)}"
```

Comment:
keypress deprecated? https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event

```suggestion
      @keydown="${(x, c) => x.keydownHandler(c.event as KeyboardEvent)}"
```

Reply:
I replaced the `keypress` event with both `keydown` and `keyup` events. The `keydown` event prevents default when the space key is pressed, and the `keyup` event triggers the click action when the space key is pressed.

This is the closest I could get to replicating the expected behavior when using `keypress`, which is what native controls still use.